src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private Node<K,V>[] nodes;
+    private int size;
+
+    public MyHashMap() {
+        nodes = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= nodes.length * LOAD_FACTOR) {
+            grow();
+        }
+        Node<K, V> newNode = nodes[getKeyPosition(key)];
+        if (newNode == null) {
+            nodes[getKeyPosition(key)] = new Node<>(key, value, null);
```

Comment:
if write like this, u shouldn't call this method twice
```suggestion
        int index = getKeyPosition(key);
        Node<K, V> newNode = nodes[index];
        if (newNode == null) {
            nodes[index] = new Node<>(key, value, null);
```